akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -36,6 +36,16 @@ import scala.concurrent.{ Future, Promise }
  */
 abstract class GraphStageWithMaterializedValue[+S <: Shape, +M] extends Graph[S, M] {
 
+  /**
+   * Grants eager access to materializer for special purposes.
+   *
+   * INTERNAL API
```

Comment:
What does it mean for this to be internal API? Can we use it from Akka HTTP? Is it tested? Will it be the default at some point?

Reply:
Btw, these are questions for @akka/akka-team not necessarily for @nvollmar :)

Also, how comes that we now have access to the materializer / actor available as eager materializer / actor when before we didn't? What changed? Couldn't we have overridden the `GraphStage.materializer` instead to enable the feature instead of adding new APIs? Was this discussed / documented?

Reply:
I don't know what your guidelines regarding this are. I was defensive on this one since it was used for a very special case and did not want to open it up for just everyone at this point. I was looking for this option in the past too, but having that option not available I usually found a better/simpler solution not requiring it.

It's used and tested by the actor ref source. If there is really a use case it could be used in IMHO.

Reply:
Some comment discussions: https://github.com/akka/akka/pull/26054#issuecomment-473854375

Reply:
It means that it's not part of the binary compatibility guarantees, i.e. we may change it. We have said that we should avoid having grey zone internal api that is used from Akka HTTP or other Akka projects outside of core Akka.

I can absolutely see how this can be useful as public api, so if we don't see any major drawback of that let's just make it so. Please create issue/pr.